packages/sdk/src/types/runtime/actor.ts

```diff
@@ -250,30 +224,15 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
 		return context.internal.CreateFromLibrary(options);
 	}
 
-	/**
-	 * Creates a new actor with a primitive shape.
-	 * @param context The SDK context object.
-	 * @param options.definiton @see PrimitiveDefinition
-	 * @param options.addCollder Whether or not to add a collider to the actor.
-	 * @param options.actor The initial state of the actor.
-	 */
-	public static CreatePrimitive(context: Context, options: {
-		definition: PrimitiveDefinition,
-		addCollider?: boolean,
-		actor?: Partial<ActorLike>
-	}): Actor {
-		return context.internal.CreatePrimitive(options);
-	}
-
 	/**
 	 * Creates a new actor hierarchy from the provided prefab.
 	 * @param context The SDK context object.
-	 * @param options.prefabId The ID of the prefab asset.
+	 * @param options.prefabId The ID of the prefab asset, or an array of assets from which the first prefab is taken.
 	 * @param options.actor The initial state of the root actor.
 	 * given a collider type when loaded @see AssetManager.loadGltf.
 	 */
 	public static CreateFromPrefab(context: Context, options: {
-		prefabId: string,
+		prefabId: string | Asset[],

```

Comment:
`prefabId` with type Asset[] is odd. How about creating an overload for it?

Reply:
How would `prefabId` work when passed an Asset array that wasn't a loaded prefab?

---
In reply to: [313508265](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r313508265) [](ancestors = 313508265)

Reply:
prefabId of type Asset[] is unintuitive. Create a function overload for this function signature.

---
In reply to: [315425105](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r315425105) [](ancestors = 315425105,313508265)

Reply:
This isn't what I expected. Is it not possible to create an overload? prefabId and firstPrefabFrom are mutually exclusive, correct?

---
In reply to: [315938355](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r315938355) [](ancestors = 315938355,315425105,313508265)

Reply:
So you want me to create an overload that takes a slightly different options object? Overloads only deal with actual arguments, not sure what it would mean to overload an options object. It'd just be the union, which is what I have.

---
In reply to: [316358367](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r316358367) [](ancestors = 316358367,315938355,315425105,313508265)

Reply:
I've added some overloads, each takes one of the args. Is that right?

---
In reply to: [316405948](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r316405948) [](ancestors = 316405948,316358367,315938355,315425105,313508265)